### PR TITLE
Update Microsoft SQL Server readme

### PR DESCRIPTION
**Pull request by ellis-elastic**
Added links to Microsoft documentation

Type of change

- Enhancement

## What does this PR do?

Added links to Microsoft documentation

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: packages/microsoft_sqlserver/_dev/build/docs/README.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
...
+The Microsoft SQL Server integration package allows you to search, observe and visualize the SQL Server audit events through Elasticsearch. 
+Auditing an instance of the SQL Server Database Engine or an individual database involves tracking and logging events that occur on the Database Engine. 
+SQL Server audit lets you create server audits, which can contain server audit specifications for server level events, and database audit specifications for database level events. 
+
+See: [SQL Server Audit page](https://docs.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-database-engine?view=sql-server-ver15) for more information on SQL Server auditing.
 
 The Microsoft SQL Server integration package allows you to search, observe and visualize the SQL Server audit events through Elasticsearch.
```
Comment (r00tu53r) [inline packages/microsoft_sqlserver/_dev/build/docs/README.md:9]: This line can be removed as it has been moved up.

File: packages/microsoft_sqlserver/docs/README.md [hunk 2]
```diff
@@ -37,7 +43,6 @@ The SQL Server audit dataset provides events from the configured Windows event l
 | destination.user.domain | Name of the directory the user is a member of. For example, an LDAP or Active Directory domain name. | keyword |
 | destination.user.id | Unique identifier of the user. | keyword |
 | destination.user.name | Short name or login of the user. | keyword |
-| destination.user.name.text | Multi-field of `destination.user.name`. | match_only_text |
```
Comment (r00tu53r) [inline packages/microsoft_sqlserver/docs/README.md:40]: Not sure why the field docs are changing. Please do a clean rebuild. The only files that must change are `_dev/build/docs/README.md, docs/README.md, changelog.yml, manifest.yml`.

**Review by r00tu53r — CHANGES_REQUESTED** [review]

File: packages/microsoft_sqlserver/docs/README.md (hunk 2 above)
Comment (ellis-elastic) [reply packages/microsoft_sqlserver/docs/README.md:40]: re ran build

File: packages/microsoft_sqlserver/_dev/build/docs/README.md (hunk 1 above)
Comment (ellis-elastic) [reply packages/microsoft_sqlserver/_dev/build/docs/README.md:7]: Updated file to remove the lines

**Review by r00tu53r — APPROVED** [review]
LGTM. @ellis-elastic thank you for the contribution. 

I noticed the CI was failing as the generated README.md (the one under `docs/README.md`) was not committed/pushed. So I pushed that.

**Pull request merged**